backend/app/services/jwt_token_service.py

```diff
@@ -0,0 +1,128 @@
+        if token_model:
+            return self.update(
+                db=self.db, obj_in=TokenUpdate(status=False), id=token_model.id
+            )
+        return False
+
+    def is_blacklist_token(self, user_id: int, token: str) -> bool:
+        token_model = self.get_by_fields(self.db, {"user_id": user_id, "token": token})
+        return token_model and not token_model.status
+
+    def create_token(
+        self,
+        email: str,
+        id: int,
+        validity: timedelta,
+        type: str = ACCESS_TOKEN,
+        role: str = USER,
+    ):
+        encode = {"sub": email, "id": id, "type": type, "role": role}
+        expires = datetime.now() + validity
+        encode.update({"exp": expires})
+        token = jwt.encode(
+            encode, settings.app.secret_key, algorithm=settings.app.algorithm
+        )
+        self.store_token(id, token)
+        return token
+
+    def verify_token(self, token: str):
+        try:
+            payload = decode_token(token)
+            username = payload.get("sub")
+            user_id = payload.get("id")
+            token_type = payload.get("type")
+
+        except ExpiredSignatureError:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=UNAUTHORIZE_MESSAGE,
+            )
+
+        except JWTError:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=UNAUTHORIZE_MESSAGE,
+            )
+
+        if not username or not user_id:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=UNAUTHORIZE_MESSAGE,
+            )
+
+        if self.is_blacklist_token(user_id, token):
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=UNAUTHORIZE_MESSAGE,
+            )
+
+        user = self.user_crud.get(self.db, user_id)
+
+        if not user:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=USER_NOT_FOUND_MESSAGE,
+            )
+
+        return {
+            "id": user_id,
+            "username": username,
+            "avatar": image_service.get_file(user.avatar),
+            "full_name": user.full_name,
+            "role": user.role.name,
+            "is_email_verified": user.is_email_verified,
+            "is_active": user.is_active,
+            "token_type": token_type,
+        }
+
+    def refresh_token(self, refresh_token: str) -> str | bool:
+        token_details = self.verify_token(refresh_token)
+
+        if token_details is None:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=UNAUTHORIZE_MESSAGE,
+            )
+        access_token = self.create_token(
+            token_details["username"], token_details["id"], timedelta(minutes=20)
+        )
```

Comment:
Hardcoded validity periods for token. Make token validity periods configurable through settings.